src/alphafold_impact/pipelines/data_collection_nih_clinical_trials/nodes.py

```diff
@@ -0,0 +1,188 @@
+"""
+This module contains functions for fetching NIH clinical trials data.
+
+Functions:
+    collect_and_save_nih_clinical_trials: Downloads, extracts, combines
+        and saves NIH clinical trials data as a partitioned dataset.
+"""
+import logging
+from pathlib import Path
+from typing import List, Dict, Callable
+import zipfile
+import shutil
+import contextlib
+import json
+
+import requests
+import pandas as pd
+
+logger = logging.getLogger(__name__)
+
+
+def _download_zip_file(zip_url: str, zip_save_path: str) -> Path:
+    """
+    Download a ZIP file from a given URL.
+
+    Args:
+        zip_url (str): URL of the zip file to be downloaded.
+        zip_save_path (str): Path to save the downloaded file.
+
+    Returns:
+        Path: The path to the downloaded zip file.
+    """
+    logger.info("Downloading ZIP file from %s", zip_url)
+    response = requests.get(zip_url, timeout=30)
+    response.raise_for_status()
+    zip_save_path = Path(zip_save_path)
+    zip_save_path.parent.mkdir(parents=True, exist_ok=True)
+    with open(zip_save_path, "wb") as file:
+        file.write(response.content)
+
+    logger.info("Downloaded ZIP file to %s", zip_save_path)
+    return zip_save_path
```

Comment:
You may want to change the name of the Path() object for it to be distinct from the path str argument. It makes it cleaner.